x/pubsub/types/proposal.go

```diff
@@ -0,0 +1,184 @@
+package types
+
+import (
+	fmt "fmt"
+	"net/url"
+
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
+)
+
+const (
+	ProposalTypeAddPublisher              = "AddPublisher"
+	ProposalTypeRemovePublisher           = "RemovePublisher"
+	ProposalTypeAddDefaultSubscription    = "AddDefaultSubscription"
+	ProposalTypeRemoveDefaultSubscription = "RemoveDefaultSubscription"
+)
+
+var _ govtypes.Content = &AddPublisherProposal{}
+var _ govtypes.Content = &RemovePublisherProposal{}
+var _ govtypes.Content = &AddDefaultSubscriptionProposal{}
+var _ govtypes.Content = &RemoveDefaultSubscriptionProposal{}
```

Comment:
what does this do?

Reply:
Default subscriptions are a gov prop to say "by default, subscribe to this publisher FQDN for this subscription ID" and we can have the client respect this or override it with local preferences. This allows the protocol to register new publisher/cellar combos without manual intervention from all validators.